src/components/common/DatePicker/index.tsx

```diff
@@ -1,59 +1,44 @@
 import React, { useState } from 'react'
-import ReactDayPicker, {
-	CaptionElementProps,
+import {
+	DayPicker,
+	DayPickerDefaultProps,
 	DayPickerProps
 } from 'react-day-picker'
 
 import { GenericComponentProps } from '../../../interfaces/GenericComponentProps'
-import YearMonthForm from './YearMonthForm'
+import LeftArrow from './icons/arrow-left.svg'
+import RightArrow from './icons/arrow-right.svg'
 
 export interface CommonDatePickerProps
 	extends GenericComponentProps,
-		Pick<
-			DayPickerProps,
-			| 'className'
-			| 'selectedDays'
-			| 'modifiers'
-			| 'onDayClick'
-			| 'locale'
-			| 'months'
-			| 'weekdaysLong'
-			| 'weekdaysShort'
-		> {
+		Pick<DayPickerDefaultProps, 'selected' | 'onDayClick'> {
 	dayPickerProps?: DayPickerProps
 	yearMonthSelect?: boolean
 }
 
 export const CommonDatePicker: React.FC<
 	CommonDatePickerProps & { initialDate: Date | null }
 > = ({ testId, ...props }) => {
-	const [month, setMonth] = useState(props.initialDate ?? new Date())
+	const [selectedDay, setSelectedDay] = useState(
+		props.initialDate ?? new Date()
+	)
 
 	return (
-		<ReactDayPicker
-			firstDayOfWeek={1}
-			captionElement={
-				props.yearMonthSelect
-					? ({
-							date,
-							localeUtils,
-							locale
-					  }: CaptionElementProps): React.ReactElement => {
-							const months = props.months ?? localeUtils.getMonths(locale)
-							return (
-								<YearMonthForm
-									date={date}
-									onChange={setMonth}
-									months={months}
-								/>
-							)
-					  }
-					: undefined
-			}
-			month={month}
-			containerProps={{
-				// @ts-ignore ReactPicker type does not allow data-* attributes
-				'data-testid': testId
+		<DayPicker
+			selected={selectedDay}
+			onDayClick={(day, { disabled }): void => {
+				if (disabled) return
+				setSelectedDay(day)
+			}}
+			data-testid={testId}
+			captionLayout={props.yearMonthSelect ? 'dropdown-buttons' : 'buttons'}
+			defaultMonth={props.initialDate ?? new Date()}
+			fromYear={props.dayPickerProps?.fromYear ?? new Date().getFullYear() - 5}
+			toYear={props.dayPickerProps?.toYear ?? new Date().getFullYear() + 5}
+			// There is an unpleasant issue with the custom arrows. On selecting date it is re-rendered and this is causing a blink. When default arrows are there, it is not happening. It would be worth considering to use default arrows.
```

Comment:
Honestly, I haven't noticed any blinking but there's a hover (circle) on the arrows and these circles are cropped out because of `overflow: hidden`. (both `DatePicker` and `DateRangePicker`)

<img width="333" alt="Screenshot 2023-06-12 at 9 57 43" src="https://github.com/purple-technology/phoenix-components/assets/4357381/75909e30-08f8-4882-8b26-85a3c1fa2aa9">

Reply:
@samuelg0rd0n I observed this styling bug for a quick moment, but when I refreshed the page it disappeared and now I'm unable to replicate it again. I tried incognito mode, but it still looks correctly. Do you still see it the way you posted?

Reply:
It looks alright now, thank you 😊